tests/jballerina-unit-test/src/test/resources/test-src/query/query-action.bal

```diff
@@ -862,6 +862,17 @@ function testJoinedQueryActionWithRegExp() {
     "(?:A|B|[ab-fgh]+(?im-x:[cdeg-k]??AAAB*[^abc-efg](?:A|B|[ab-fgh]+(?im-x:[cdeg-k]??A)|)|^|PQ?)|)|^|PQ?");
 }
 
+function testQueryActionWithCollectClauseInsideLeClause() {
```

Comment:
Shall we add one test with similar to the issue,
with an empty do clause.

Reply:
Added in commit [#f0b8530](https://github.com/ballerina-platform/ballerina-lang/pull/41413/commits/f0b85305ea284059a10559877e20523b3a47cf94)